src/trie/path.rs

```diff
@@ -104,25 +104,25 @@ impl Path {
     }
 }
 
-impl From<Path> for TrieKey {
+impl From<Path> for Vec<u8> {
```

Comment:
yup, rather than removing the TrieKey and having everything be a `Vec<u8>` I would prefer add more distinct types and be explicit about which one should be used at each step and how they combine

Reply:
I just added a comment because I don't want this conversion to be "key specific" but used for different use cases where you need `Path` as `Vec<u8>`